File: docs/snippets/0027_remove_element.py
from typing import List


# Fast Slow Pointers
def removeElement(nums: List[int], val: int) -> int:
    fast, slow = 0, 0

    while fast < len(nums):
        if nums[fast] != val:
            nums[slow] = nums[fast]
            slow += 1
        fast += 1

    return slow


nums = [0, 1, 2, 2, 3, 0, 4, 2]
val = 2
print(removeElement(nums, val))  # 5

File: docs/snippets/cpp/0027_remove_element.cc
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int slow = 0;

        for (size_t fast = 0; fast < nums.size(); fast++)
        {
            if (val != nums[fast])
            {
                nums[slow++] = nums[fast];
            }
        }
        return slow;
    }
};

int main()
{
    Solution solution;
    vector<int> nums = {3, 2, 2, 3};
    int val = 3;
    cout << solution.removeElement(nums, val) << endl; // 2
    return 0;
}
